src/components/TwoOptionsToggle.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+import { COLORS } from '../styles/themes';
+
+/**
+ * @param {{
+ *   options: Map<'first'|'second', { value: string; onTap: Function; }>;
+ *   defaultOption: 'first'|'second';
+ * }}
+ */
+export const TwoOptionsToggle = ({ options, defaultOption }) => {
+  const [currOption, setCurrOption] = useState(defaultOption);
+  const isActive = (option) => (componentOption) => option === componentOption;
+  const onTap = (option) => {
+    setCurrOption(option);
+    // Execute the callback assigned to the option
+    options[option].onTap();
+  };
+  return (
+    <View style={[styles.wrapper]}>
+      <Option option='first' optionValue={options.first.value} isActive={isActive(currOption)} onTap={onTap} />
```

Comment:
Why do you need to have `currOption` as parameter if you can always get it from the state?

Reply:
`Option` is also a component and I did it in a general way. `isActive` is a currying function, which means the first call holds the "current option" selected by default or by a user and return another function which are able to answer if the "current option" match the "option value" of `Option` component.

Reply:
Oh, I did this way because I was following a principle: "independent and composable components". If they are controlled from outside they become attached to the parent. I did all the code this way, as much as I could.